modules/authentication/src/handlers/local.ts

```diff
@@ -200,20 +201,43 @@ export class LocalHandlers {
         .add(config.tokenInvalidationPeriod as number, 'milliseconds')
         .toDate(),
     });
+    let refreshToken: RefreshToken | null;
+    if (config.generateRefreshToken) {
+      refreshToken = await RefreshToken.getInstance().create({
+        userId: user._id,
+        clientId,
+        token: AuthUtils.randomToken(),
+        expiresOn: moment()
+          .add(config.refreshTokenInvalidationPeriod as number, 'milliseconds')
+          .toDate(),
+      });
+    }
 
-    const refreshToken: RefreshToken = await RefreshToken.getInstance().create({
-      userId: user._id,
-      clientId,
-      token: AuthUtils.randomToken(),
-      expiresOn: moment()
-        .add(config.refreshTokenInvalidationPeriod as number, 'milliseconds')
-        .toDate(),
-    });
+    if (config.setCookies.enabled) {
+      const cookieOptions = config.setCookies.options;
+      const cookies: Cookie[] = [{
+        name: 'accessToken',
+        value: (accessToken as AccessToken).token,
+        options: cookieOptions
+
+      }];
+      if (!isNil(refreshToken!)) {
+        cookies.push({
+          name: 'refreshToken',
+          value: refreshToken.token,
+          options: cookieOptions
+        });
+      }
+      return {
+        result: { message: 'Successfully authenticated' },
+        setCookies: cookies,
+      };
+    }
 
     return {
       userId: user._id.toString(),
       accessToken: accessToken.token,
-      refreshToken: refreshToken.token,
+      refreshToken: !isNil(refreshToken!) ? refreshToken!.token : undefined,
```

Comment:
you're checking if the refreshToken is null, and at the same time you're using "refreshToken!" these are contradictory.